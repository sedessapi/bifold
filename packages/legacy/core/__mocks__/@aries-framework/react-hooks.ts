/* eslint-disable @typescript-eslint/no-explicit-any */
import { LegacyIndyCredentialFormat } from '@aries-framework/anoncreds'
import {
  BasicMessageRecord,
  CredentialExchangeRecord,
  CredentialProtocolOptions,
  ProofExchangeRecord,
} from '@aries-framework/core'

const useCredentials = jest.fn().mockReturnValue({ records: [] } as any)
const useProofs = jest.fn().mockReturnValue({ records: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialExchangeRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofExchangeRecord[])
const useBasicMessagesByConnectionId = jest.fn().mockReturnValue([] as BasicMessageRecord[])
const useBasicMessages = jest.fn().mockReturnValue({ records: [] as BasicMessageRecord[] })
const mockCredentialModule = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
  getFormatData: jest
    .fn()
    .mockReturnValue(
      Promise.resolve({} as CredentialProtocolOptions.GetCredentialFormatDataReturn<[LegacyIndyCredentialFormat]>)
    ),
}
const mockProofModule = {
  getCredentialsForRequest: jest.fn(),
  acceptRequest: jest.fn(),
  declineRequest: jest.fn(),
  getFormatData: jest.fn(),
  findRequestMessage: jest.fn(),
  requestProof: jest.fn(),
  update: jest.fn(),
}

const mockMediationRecipient = {
  initiateMessagePickup: jest.fn(),
}
const mockOobModule = {
  findById: jest.fn().mockReturnValue(Promise.resolve(null)),
  createInvitation: jest.fn(),
  toUrl: jest.fn(),
}
const mockBasicMessageRepository = {
  update: jest.fn(),
}
const mockAgentContext = {
  dependencyManager: {
    resolve: jest.fn().mockReturnValue(mockBasicMessageRepository),
  },
}
const useAgent = () => ({
  agent: {
    credentials: mockCredentialModule,
    proofs: mockProofModule,
    mediationRecipient: mockMediationRecipient,
    oob: mockOobModule,
    context: mockAgentContext,
  },
})
const useCredentialById = jest.fn()
const useProofById = jest.fn()
const useConnectionById = jest.fn()
const useConnections = jest.fn()

export {
  useAgent,
  useConnectionById,
  useCredentials,
  useProofs,
  useCredentialById,
  useCredentialByState,
  useProofById,
  useProofByState,
  useConnections,
  useBasicMessages,
  useBasicMessagesByConnectionId,
}
