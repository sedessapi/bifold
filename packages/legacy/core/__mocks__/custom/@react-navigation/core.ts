const navigate = jest.fn()
const dispatch = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  setOptions: jest.fn(),
  getParent: () => {return {
    navigate,
    dispatch,
  }},
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
