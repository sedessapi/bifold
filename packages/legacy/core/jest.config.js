module.exports = {
  preset: 'react-native',
  testTimeout: 10000,
  setupFiles: ['<rootDir>/jestSetup.js'],
  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  moduleNameMapper: {
    '\\.(jpg|ico|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mocks__/file.js',
    '\\.(css|less)$': '<rootDir>/__mocks__/style.js',
    axios: require.resolve('axios'),
    'react-i18next': '<rootDir>/__mocks__/react-i18next.ts',
    '^uuid$': require.resolve('uuid'),
    '@aries-framework/core': require.resolve('@aries-framework/core'),
    '@aries-framework/anoncreds': require.resolve('@aries-framework/anoncreds'),
  },
  transform: {
    '^.+\\.(js|jsx|ts|tsx)$': 'babel-jest',
  },
  transformIgnorePatterns: [
    'node_modules\\/(?!(.*react-native.*)|(uuid)|(@aries-framework\\/core)|(@aries-framework\\/anoncreds))',
  ],
  testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.[jt]sx?$',
  testPathIgnorePatterns: [
    '\\.snap$',
    '<rootDir>/node_modules/',
    '<rootDir>/lib',
    '<rootDir>/__tests__/contexts/',
    '<rootDir>/__tests__/helpers/',
  ],
  coveragePathIgnorePatterns: [
    '/node_modules/',
    '/__tests__/',
    '<rootDir>/App/navigators/defaultStackOptions.tsx',
    '<rootDir>/App/defaultConfiguration.ts',
    '<rootDir>/App/components/buttons/InfoIcon.tsx',
    '<rootDir>/App/hooks/deep-links.ts',
  ],
  cacheDirectory: '.jest/cache',
  snapshotFormat: {
    escapeString: true,
    printBasicPrototype: true,
  },
}
