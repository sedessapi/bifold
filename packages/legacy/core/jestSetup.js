/* eslint-disable no-undef */
// eslint-disable-next-line import/no-extraneous-dependencies
import 'reflect-metadata'
import 'react-native-gesture-handler/jestSetup'
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock.js'
import mockRNLocalize from 'react-native-localize/mock'

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('react-native-localize', () => mockRNLocalize)
