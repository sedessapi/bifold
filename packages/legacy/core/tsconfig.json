{
  "extends": "../../../tsconfig-base.json",
  "compilerOptions": {
    "baseUrl": "./App",
    "outDir": "./lib"
  },
  "exclude": ["node_modules", "babel.config.js", "jest.config.js", "jestSetup.js", "lib"]
}