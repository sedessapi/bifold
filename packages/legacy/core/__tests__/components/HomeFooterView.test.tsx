import { CredentialExchangeRecord as CredentialRecord, CredentialState } from '@aries-framework/core'
import { useCredentialByState } from '@aries-framework/react-hooks'
import { render } from '@testing-library/react-native'
import React from 'react'

import HomeFooterView from '../../App/components/views/HomeFooterView'

describe('HomeFooterView Component', () => {
  test('Renders correctly with no notifications', async () => {
    const tree = render(<HomeFooterView />)

    expect(tree).toMatchSnapshot()
  })

  test('Renders correctly with notifications', async () => {
    const testCredentialRecords: CredentialRecord[] = [
      new CredentialRecord({
        threadId: '1',
        state: CredentialState.OfferReceived,
        protocolVersion: 'v1',
      }),
    ]
    // @ts-ignore
    useCredentialByState.mockReturnValue(testCredentialRecords)

    const tree = render(<HomeFooterView />)

    expect(tree).toMatchSnapshot()
  })
})
