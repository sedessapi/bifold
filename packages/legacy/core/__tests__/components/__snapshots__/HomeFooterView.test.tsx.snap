// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeFooterView Component Renders correctly with no notifications 1`] = `
<View>
  <View
    style={
      Object {
        "paddingBottom": 75,
        "paddingHorizontal": 25,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginTop": 35,
          },
        ]
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 38,
              "fontWeight": "bold",
            },
            Object {
              "marginBottom": 20,
              "marginTop": 25,
            },
          ]
        }
      >
        Home.Welcome
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginTop": 35,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "marginTop": 25,
              "textAlign": "center",
            },
          ]
        }
      >
        Home.NoCredentials
      </Text>
    </View>
  </View>
</View>
`;

exports[`HomeFooterView Component Renders correctly with notifications 1`] = `
<View>
  <View
    style={
      Object {
        "paddingBottom": 75,
        "paddingHorizontal": 25,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginTop": 35,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "marginTop": 25,
              "textAlign": "center",
            },
          ]
        }
      >
        <Text>
          Home.YouHave
           
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            2
          </Text>
           
          Home.Credentials
           
          Home.InYourWallet
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
